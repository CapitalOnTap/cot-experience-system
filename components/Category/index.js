import { Title } from 'cot-experience';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const ColorBox = styled.div`
  width: 24px;
  height: 24px;
  background: ${props => props.color};
  margin-right: 8px;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Wrapper = styled.div`
  display: flex;
`;

const Category = ({ name, color, ...props }) => (
  <Wrapper>
    <ColorBox color={color}>{name.split('')[0]}</ColorBox>
    <Title>{name}</Title>
  </Wrapper>
);

Category.propTypes = {
  name: PropTypes.string,
  color: PropTypes.string,
};

Category.defaultProps = {
  name: 'Category name',
  color: '#27B161',
};

export default Category;
