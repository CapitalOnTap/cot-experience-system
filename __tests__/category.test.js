/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Category from '../components/Category';

describe('With Enzyme', () => {
  it('It renders', () => {
    const wrapper = shallow(<Category />);

    expect(wrapper.length).toBe(1);
  });
});

describe('With Snapshot Testing', () => {
  it('It renders"', () => {
    const component = renderer.create(<Category />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
