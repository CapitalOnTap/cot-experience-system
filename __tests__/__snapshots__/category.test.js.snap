// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing It renders" 1`] = `
<div
  className="Category__Wrapper-sc-1aaoz62-1 cbNQNy"
>
  <div
    className="Category__ColorBox-sc-1aaoz62-0 kBUVu"
    color="#27B161"
  >
    C
  </div>
  <h5
    className="sc-jTzLTM kVtfjo"
  >
    Category name
  </h5>
</div>
`;
